### DISCUSSION TITLE
Cheat Tracker

### DISCUSSION THREAD

**Discussion by SporyTike** [RetroAchievements/RAWeb#846] [Ideas]
Many old games have inbuild cheats or debug menus which can be activated by pressing button combinations or entering specific passwords. Some games can't even have real sets because they can easily be cheated completely by using these "features" resulting in free points and mastery badges.
The only way developers currently can prevent the usage is to block everything from working as soon they were used, but this only works for the current session. So games like RPGs where you can cheat stats, then save and reload the game are basically impossible to protect without spamming an achievement every few seconds.

I thought about a tracking-system what would allow us to handle this issue:
- A flag-system which uses similar code to leaderboards and achievements, when a specific condition is met, it submits the timestamp when it happened and the flag itself
- The tracking can't be disabled in the emulators
- Only specific roles can see information about the tracking-system (e.g. cheat-investigation team, admins, mods)
--- If a tracking-system exists
--- Who got tracked and when
--- Small description field what exactly activates the system
- Only specific roles can maintain the tracking-system
--- Adding them
--- Editing them
--- Removing them
--- Reset the flag for specific users (e.g. when they reset their progress)
- The user should be informed about their flagging
--- A popup on the emulator
--- A warning on the game page including
------ Consequences of the flag
------ How to undo it / who to contact

Ideas which needs definitely more discussion first:
- Being flagged as cheater has the following consequences
--- Achievements unlocked after the flag got set do not count to the points the user has until the flag gets removed
--- Achievements unlocked after the flag got set do not count to the mastery until the flag gets removed

I know that this is not only web territory, also toolkit, but I am not familiar enough with github so I don't know how to link both of the branches with this issue, sorry for that.

**Comment by Blazekickn** [comment]
I'm not a huge fan of being spied on while I play. Multisession would handle this just fine (besides the reporting to admin team) and would be multipurpose. I also think if this does get implemented developers should be at least allowed to see whether the game has a tracking system if they can't add it themselves, since they're probably going to be asking staff to put it on their sets.

also before you ask yes i just decided to check the github for no reason

**Reply by MagmarFire** [reply]
Perhaps we should have something in the terms of service that says that by playing RetroAchievements, you agree to send information with the site. Privacy is a big concern, but cheating is also a serious problem. Players should understand that devs have a stake in maintaining the integrity of earn rates as much as possible.

There's also, of course, the whole Rich Presence thing, which is also used by some devs as a cheat flagger.

**Reply by Blazekickn** [reply]
being locked out of the entire set until you can get an admin to undo the flag (most likely after telling you to reset everything) because you used a cheat code once just seems a little overkill imo. There are also plenty of cases where this form of cheat protection wouldn't really work - like say I'm making a bonus set and one of the achievements is a full game run with some restriction. I'd want to to lock that single achievement from triggering during that playthrough, but other achievements should still be allowed to trigger. With this system I can't really do that,

**Reply by Blazekickn** [reply]
yes i understand you would be locked out of the entire set with multisession as well, but there you can set up code to remove the cheat protection if you start a new file or delete your old one. You don't have to beg staff to let you play the game again.

**Comment by televandalist** [comment]
I think disabling the ability to save while certain conditions are true would take care of a lot of this.

**Comment by Lanzador** [comment]
Meh. I agree that this is too much. I dislike the idea of some dangerous trigger that just bans you from the game until your case is manually checked. The fact you used a cheat code doesn't even necessarily mean you wanted to cheat the achievements.
